xpath-outcome-initiator/src/main/java/org/cristalise/kernel/persistency/outcomebuilder/field/DecimalField.java

```diff
@@ -46,4 +50,46 @@ public void setValue(Object value) throws InvalidOutcomeException {
 
         super.setData(value.toString());
     }
+
+    @Override
+    public void setNgDynamicFormsValidators(JSONObject validators) {
+        //locale specific separators could be used, but it should be based on the locale of the browser
+        //char separator = new DecimalFormatSymbols(Locale.getDefault(Locale.Category.FORMAT)).getDecimalSeparator();
+        char separator = Gateway.getProperties().getString("Webui.decimal.separator", ".").charAt(0);
+
+        super.setNgDynamicFormsValidators(validators);
+
+        if (StringUtils.isNotBlank(pattern)) {
+            validators.put("pattern", pattern);
+        }
+        else {
+            Integer totalDigits = null;
+            Integer fractionDigits = null;
+
+            if(contentType.hasFacet(Facet.TOTALDIGITS)) {
+                Facet f = contentType.getFacet(Facet.TOTALDIGITS);
+                totalDigits = Integer.valueOf(f.getValue());
+            }
+
+            if(contentType.hasFacet(Facet.FRACTIONDIGITS)) {
+                Facet f = contentType.getFacet(Facet.FRACTIONDIGITS);
+                fractionDigits = Integer.valueOf(f.getValue());
+            }
+
+            if (totalDigits == null && fractionDigits == null) {
+                //default validator for any decimal field
+                validators.put("pattern", "^\\d+\\" + separator + "?\\d+$");

```

Comment:
This requires at least 2 digits. Use the pattern I suggested.

Reply:
as this is a default pattern, I do not see why the "at least the 2 digits" is required

Reply:
Because you have 2 times '\\d+' with an optional separator between them. Each of those requires one digit. Test it: https://jsfiddle.net/wphb21vf/

Reply:
I do not see why the default pattern should restrict decimal numbers to 'at least 2 fraction digits'?!?! It shall only check that the number is decimal.

Reply:
I think this is the good pattern to be used: 

`^(?!-0(\.0+)?$)-?(0|[1-9]\d*)(\.\d+)?$`

Check it here: https://regex101.com/r/kD0cE3/1